hazelcast/src/test/java/com/hazelcast/cluster/TcpIpMemberConnectionLossTest.java

```diff
@@ -0,0 +1,155 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.cluster;
+
+import com.hazelcast.config.Config;
+import com.hazelcast.config.JoinConfig;
+import com.hazelcast.config.NetworkConfig;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.instance.impl.HazelcastInstanceFactory;
+import com.hazelcast.test.Accessors;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.annotation.SlowTest;
+import org.jetbrains.annotations.NotNull;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+import static com.hazelcast.spi.properties.ClusterProperty.CONNECTION_MONITOR_INTERVAL;
+import static com.hazelcast.test.HazelcastTestSupport.assertClusterSize;
+import static com.hazelcast.test.HazelcastTestSupport.assertTrueEventually;
+import static com.hazelcast.test.HazelcastTestSupport.randomName;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category(SlowTest.class)
+public class TcpIpMemberConnectionLossTest {
+
+    @Before
+    @After
+    public void killAllHazelcastInstances() throws IOException {
+        HazelcastInstanceFactory.shutdownAll();
+    }
+
+    @Test
+    public void test_whenCanNotReconnect() {
+        Config config = getConfig();
+
+        HazelcastInstance h1 = Hazelcast.newHazelcastInstance(config);
+        HazelcastInstance h2 = Hazelcast.newHazelcastInstance(config);
+        assertClusterSize(2, h1, h2);
+
+        //when a member disappears and the other one can't connect to it
+        h1.shutdown();
+        //then it eventually gets dropped from the cluster
+        assertTrueEventually(() -> assertClusterSize(1, h2));
+    }
+
+    @Test
+    public void test_whenConnectionKeepsDying() throws Exception {
+        Config config = getConfig();
+
+        HazelcastInstance h1 = Hazelcast.newHazelcastInstance(config);
+        HazelcastInstance h2 = Hazelcast.newHazelcastInstance(config);
+        assertClusterSize(2, h1, h2);
+
+        InetSocketAddress address = Accessors.getNodeEngineImpl(h1).getThisAddress().getInetSocketAddress();
+        RejectionProxy rejectionProxy = new RejectionProxy(address.getHostName(), address.getPort());
+        try {
+            //when a member disappears
+            h1.shutdown();
```

Comment:
`shutdown()` is graceful ie notifies master, ensures migrations are done then closes connections and shuts down. I think you may want to use `h1.getLifecycleService().terminate();` here (and in previous test) instead.